### PR TITLE
Compile with gcc 4.8

### PR DESCRIPTION
**Pull request by franksinankaya**
@am11 @janvorli @jkotas

### CODE REVIEW

**Comment by franksinankaya** [comment]
We should be able to close #26929 and #25529 after this change.

File: src/vm/diagnosticsprotocol.h [hunk 1]
```diff
@@ -210,7 +210,7 @@ namespace DiagnosticsIpc
 
         // Initialize an outgoing IpcMessage with a header and payload
         template <typename T>
-        bool Initialize(IpcHeader header, T&& payload)
+        bool InitializePayload(IpcHeader header, T&& payload)
```
Comment (jkotas) [inline src/vm/diagnosticsprotocol.h]: I think it would be better to delete this method and just use the other overload.

**Comment by jkotas** [comment]
LGTM otherwise

File: src/vm/diagnosticsprotocol.h (hunk 1 above)
Comment (franksinankaya) [reply src/vm/diagnosticsprotocol.h]: That was my reaction too.

Code is sending a S_OK constant as in src/vm/dumpdiagnosticprotocolhelper.cpp below. I couldn't get that to compile Any suggestions?

Comment (jkotas) [reply src/vm/diagnosticsprotocol.h]: HRESULT success = S_OK;
... pass success to the method ...

Comment (franksinankaya) [reply src/vm/diagnosticsprotocol.h]: will do, thanks

File: configurecompiler.cmake [hunk 2]
```diff
@@ -20,6 +20,7 @@ endif()
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
 check_cxx_compiler_flag(-faligned-new COMPILER_SUPPORTS_F_ALIGNED_NEW)
+check_cxx_compiler_flag(-fstack-protector-strong COMPILER_SUPPORTS_F_STACK_PROTECTOR_STRONG)
```
Comment (am11) [inline configurecompiler.cmake]: nit: these two lines can go near where they are used (just before their if conditions; they are used only once).

**Comment by xiangzhai** [comment]
Hi @franksinankaya 

Will backport the patch to [3.1 branch](https://github.com/dotnet/coreclr/tree/release/3.1)?

Thanks,
Leslie Zhai

**Comment by franksinankaya** [comment]
3.1 branch unfortunately won't work because full GCC compilation has been only enabled on master branch only recently.

**Pull request merged**